src/main/cs/Infinispan/HotRod/Impl/RemoteCacheSWIGImpl.cs

```diff
@@ -247,17 +250,22 @@ public QueryResponse Query(QueryRequest qr)
             return QueryResponse.Parser.ParseFrom(respBytes);
         }
 
-        public byte[] Execute(string scriptName, IDictionary<string, string> dict)
+        public object Execute(string scriptName, IDictionary<string, string> dict = null)
```

Comment:
Hmm, so the Execute method only accepts a dictionary of Strings (both keys and values). It would be really good to accept Strings for keys and any type of object for values. Otherwise being able to pass a specific marshaller for script arguments seems useless - Strings are always used and so the internal marshaller is enough.

Reply:
Working on this